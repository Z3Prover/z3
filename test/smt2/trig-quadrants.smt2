;; Copyright (c) 2016 Microsoft Corporation
(set-info :source |Written by D. B. Staple for GitHub issue #680.|)
(set-info :status sat)

(set-option :pp.decimal true)
(set-option :model_validate true)

; pi / 4
(define-fun sqrt2over2 () Real (root-obj (+ (* 2 (^ x 2)) (- 1)) 2))
(declare-fun theta () Real)
(assert (= (cos theta) sqrt2over2))
(assert (= (sin theta) sqrt2over2))
(check-sat-using qfnra-nlsat)
(get-model)
(reset)

; 3 * pi / 4
(define-fun sqrt2over2 () Real (root-obj (+ (* 2 (^ x 2)) (- 1)) 2))
(declare-fun theta () Real)
(assert (= (cos theta) (- sqrt2over2)))
(assert (= (sin theta) sqrt2over2))
(check-sat-using qfnra-nlsat)
(get-model)
(reset)

; 5 * pi / 4
(define-fun sqrt2over2 () Real (root-obj (+ (* 2 (^ x 2)) (- 1)) 2))
(declare-fun theta () Real)
(assert (= (cos theta) (- sqrt2over2)))
(assert (= (sin theta) (- sqrt2over2)))
(check-sat-using qfnra-nlsat)
(get-model)
(reset)

; 7 * pi / 4
(define-fun sqrt2over2 () Real (root-obj (+ (* 2 (^ x 2)) (- 1)) 2))
(declare-fun theta () Real)
(assert (= (cos theta) sqrt2over2))
(assert (= (sin theta) (- sqrt2over2)))
(check-sat-using qfnra-nlsat)
(get-model)
