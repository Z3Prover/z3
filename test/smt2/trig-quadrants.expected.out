sat
(model 
  (define-fun theta () Real
    (acos 0.7071067811?))
)
sat
(model 
  (define-fun theta () Real
    (acos (- 0.7071067811?)))
)
sat
(model 
  (define-fun theta () Real
    (+ (acos 0.7071067811?) pi))
)
sat
(model 
  (define-fun theta () Real
    (+ (acos (- 0.7071067811?)) pi))
)
